interfaces/machine_parameters/ring_section.py

```diff
@@ -192,16 +192,17 @@ def __init__(self, length, alpha_0,
 
             # orbit_bump was checked
             # The length attribute is created with no further checks
-            if orbit_bump.timebase == 'by_time':
+            self.length = self.orbit_bump + self.length_design
+            # if orbit_bump.timebase == 'by_time':
 
-                self.length = ring_programs.orbit_length_program(
-                    [self.length_design] * len(self.orbit_bump[0, 0, :]),
-                    time=self.orbit_bump[0, 0, :])
-                self.length[:, 1, :] += self.orbit_bump[:, 1, :]
+            #     self.length = ring_programs.orbit_length_program(
+            #         [self.length_design] * len(self.orbit_bump[0, 0, :]),
+            #         time=self.orbit_bump[0, 0, :])
+            #     self.length[:, 1, :] += self.orbit_bump[:, 1, :]
 
-            else:
+            # else:
 
-                self.length = self.orbit_bump + self.length_design
+            #     self.length = self.orbit_bump + self.length_design
```

Comment:
@alasheen is this suitable replacement for what you had in mind here?

Reply:
Yes that was the idea!

There are a few if statement like this one that could be replaced by handling the operation in datatypes for different `timebase`